### PR TITLE
Change angular panel display behavior

### PR DESCRIPTION
**Pull request by anthonydresser**
Changes how the angular panel decides when to display another angular component.

This requires some pretty significant changes to multiple areas using the angular panel that were not properly letting angular handle lifecycle. Going forward we should ride on the side of letting angular handle things rather than directly accessing components (this is recommended code practice by angular anyways) where we use angular. This should prevent more unintended behavior change in the future.

### CODE REVIEW

File: src/sql/parts/query/views/queryOutput.component.ts [hunk 1]
```diff
@@ -32,18 +32,20 @@ declare type PaneType = 'messages' | 'results';
...
+	@ViewChild(QueryComponent) queryComponent: QueryComponent;
 
-	@ViewChild('queryPlanComponent') queryPlanComponent: QueryPlanComponent;
+	@ViewChild(QueryPlanComponent) queryPlanComponent: QueryPlanComponent;
 
-	@ViewChild('topOperationsComponent') topOperationsComponent: TopOperationsComponent;
+	@ViewChild(TopOperationsComponent) topOperationsComponent: TopOperationsComponent;
 
-	@ViewChild('chartViewerComponent') chartViewerComponent: ChartViewerComponent;
+	@ViewChild(ChartViewerComponent) chartViewerComponent: ChartViewerComponent;
```
Comment (AbbiePetcht) [inline src/sql/parts/query/views/queryOutput.component.ts]: >@ViewChild(ChartViewerComponent) chartViewerComponent: ChartViewerComponent; [](start = 1, length = 76)

You can delete this. This parameter is unused

File: src/sql/base/browser/ui/panel/media/panel.css [hunk 2]
```diff
@@ -97,16 +104,16 @@ panel {
...
-	flex: 1 1 auto;
+	flex: 0 0 auto;
 }
 
 .tab > .tabLabel.active {
 	border-bottom: 1px solid;
 }
 
-.composite.title ~ tab.fullsize > :first-child {
+/* .composite.title ~ tab.fullsize > :first-child {
```
Comment (AbbiePetcht) [inline src/sql/base/browser/ui/panel/media/panel.css]: remove this?

File: src/sql/base/browser/ui/panel/tab.component.ts [hunk 3]
```diff
@@ -13,23 +13,33 @@ export abstract class TabChild {
...
-		<div class="visibility" [class.hidden]="!active && visibilityType == 'visibility'" *ngIf="visibilityType == 'visibility' || active" class="fullsize">
-			<ng-content class="body fullsize"></ng-content>
+		<div class="visibility" [class.hidden]="shouldBeHidden()" *ngIf="shouldBeIfed()" class="fullsize">
+			<ng-container *ngTemplateOutlet="templateRef"></ng-container>
 		</div>
 	`
 })
 export class TabComponent implements OnDestroy {
 	@ContentChild(TabChild) private _child: TabChild;
+	@ContentChild(TemplateRef) templateRef;
```
Comment (AbbiePetcht) [inline src/sql/base/browser/ui/panel/tab.component.ts]: >@ContentChild(TemplateRef) templateRef; [](start = 1, length = 39)

what do you need this for?

File: src/sql/base/browser/ui/panel/panel.component.ts [hunk 4]
```diff
@@ -67,24 +68,24 @@ let idPool = 0;
 		</div>
 	`
 })
-export class PanelComponent extends Disposable implements AfterContentInit, OnInit, OnChanges, OnDestroy, AfterViewInit {
+export class PanelComponent extends Disposable {
 	@Input() public options: IPanelOptions;
 	@Input() public actions: Array<Action>;
 	@ContentChildren(TabComponent) private _tabs: QueryList<TabComponent>;
 	@ViewChildren(TabHeaderComponent) private _headerTabs: QueryList<TabHeaderComponent>;
+	@ViewChild(ScrollableDirective) private scrollable: ScrollableDirective;
```
Comment (AbbiePetcht) [inline src/sql/base/browser/ui/panel/panel.component.ts]: >scrollable [](start = 41, length = 10)

should have _ prefix

File: src/sql/base/browser/ui/panel/tab.component.ts (hunk 3 above)
Comment (anthonydresser) [reply src/sql/base/browser/ui/panel/tab.component.ts]: its used in the template

**Pull request merged**